AfterClass-io/afterclass.io-v2
# feat(db): Prisma schema and example CRUD operations

closes #5

## Changes
- Prisma schema definition
- Supabase database migration files
- Example CRUD operations for the `Universities` table
- Removed AuthDemo and related components
- Removed `example.ts` and related components which contains the example tRPC call

### Notes
- `zod` is used for DB query and input validation instead of the prisma schema due the reasons stated in this GitHub discussion [here](https://github.com/trpc/trpc/discussions/3707)

## Screenshots
<img width="1447" alt="Screenshot 2023-12-27 at 12 47 08 AM" src="https://github.com/AfterClass-io/afterclass.io-v2/assets/68802506/9877ced8-50d7-46cc-b5d5-688a439257c1">

## Testing
- Ensure Supabase's staging project contains the right entities (there should be one pre-populated row for the Universities table, created using the `createUniversity` call which is commented out for now
- Go to the preview link and you should see "Hello from SMU. Refer https://www.smu.edu.sg" as shown in the screenshot
